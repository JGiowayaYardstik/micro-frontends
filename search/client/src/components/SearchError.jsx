import React from 'react';

const SearchError = () => {
  return <p>{'Ups, leider ist ein Fehler aufgetreten'}</p>;
};

export default SearchError;
