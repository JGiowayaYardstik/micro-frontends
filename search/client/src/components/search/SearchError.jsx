import React from 'react';
import PropTypes from 'prop-types';
import { withI18n } from 'react-i18next';
import { compose } from 'recompose';
import { withStyle, connect } from '@dm/style-provider';

const setMMBehaviourVariables = () => {
	if (typeof window.mmBehaviourUtils !== 'undefined') {
		const productOverviewNavigationSummary = {};
		productOverviewNavigationSummary.currentPage = 0;
		productOverviewNavigationSummary.productsOnPage = 0;

		window.mmBehaviourUtils.updateVariable(
			'productOverviewNavigation',
			productOverviewNavigationSummary
		);
	}
};

const SearchError = props => {
	setMMBehaviourVariables();
	return (
		<div data-dmid="search-error-wrapper" className={props.styles.searchErrorWrapper}>
			<h1 data-dmid="search-error" className={props.styles.searchError}>
				{props.t('search.error.headline')}
			</h1>
		</div>
	);
};

const searchErrorWrapper = ({ theme }) => ({
	paddingTop: theme.dimension.spacing.xxxl.rem
});

const searchError = ({ theme }) => ({
	fontFamily: theme.typography.fontFamily.primary.medium,
	fontSize: theme.typography.fontSize.xxl.rem,
	color: theme.palette.color2.hex,
	textAlign: 'center',
	margin: `${theme.dimension.spacing.m.rem} 0`
});

SearchError.propTypes = {
	styles: PropTypes.object,
	t: PropTypes.func
};

export default compose(withStyle, connect({ searchError, searchErrorWrapper }), withI18n())(
	SearchError
);
