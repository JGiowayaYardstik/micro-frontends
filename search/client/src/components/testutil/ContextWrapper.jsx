import React from 'react';
import PropTypes from 'prop-types';
import { StyleProvider } from '@dm/style-provider';
import { DeviceProvider } from '@dm/device-provider';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { theme as dm } from '@dm/ui-theme';

export default function ContextWrapper({ children, locale }) {
	return (
		<StyleProvider>
			<DeviceProvider>
				<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
					{children}
				</I18nextProvider>
			</DeviceProvider>
		</StyleProvider>
	);
}

ContextWrapper.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string
};

export function ContextWrapperWithoutResponsive({ children, locale }) {
	return (
		<StyleProvider>
			<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
				{children}
			</I18nextProvider>
		</StyleProvider>
	);
}

ContextWrapperWithoutResponsive.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string
};

export function ContextWrapperWithStore({ children, locale, theme = dm, store = global.store }) {
	return (
		<Provider store={store}>
			<StyleProvider theme={theme}>
				<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
					{children}
				</I18nextProvider>
			</StyleProvider>
		</Provider>
	);
}

ContextWrapperWithStore.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string,
	store: PropTypes.object,
	theme: PropTypes.object
};
