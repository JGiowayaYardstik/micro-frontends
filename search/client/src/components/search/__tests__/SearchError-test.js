import React from 'react';
import { createSnapshot } from '@dm/jest-utils';
import ContextWrapper from '../../testutil/ContextWrapper';

import SearchError from '../SearchError';

describe('SearchError', () => {
  it('renders and matches snapshot (DE)', () => {
    const snapshot = createSnapshot(
      <ContextWrapper locale="de">
        <SearchError />
      </ContextWrapper>
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('renders and matches snapshot (HU)', () => {
    const snapshot = createSnapshot(
      <ContextWrapper locale="hu">
        <SearchError />
      </ContextWrapper>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
