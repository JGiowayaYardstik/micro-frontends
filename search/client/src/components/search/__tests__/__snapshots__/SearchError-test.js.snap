// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchError renders and matches snapshot (DE) 1`] = `
Object {
  "_locale": "de",
  "_width": 1200,
  "html": "
<div data-dmid=search-error-wrapper className=searchErrorWrapper__h06092>
  <h1 data-dmid=search-error className=searchError__bc9skf>
    Leider ist ein technischer Fehler aufgetreten.
  </h1>
</div>;
",
  "styles": "
.searchError__bc9skf {
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  font-size: 1.75rem;
  color: #002878;
  text-align: center;
  margin: 1.5625rem 0;
}
.searchErrorWrapper__h06092 {
  padding-top: 3.75rem;
}
",
}
`;

exports[`SearchError renders and matches snapshot (HU) 1`] = `
Object {
  "_locale": "de",
  "_width": 1200,
  "html": "
<div data-dmid=search-error-wrapper className=searchErrorWrapper__h06092>
  <h1 data-dmid=search-error className=searchError__bc9skf>
    Technikai hiba történt.
  </h1>
</div>;
",
  "styles": "
.searchError__bc9skf {
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  font-size: 1.75rem;
  color: #002878;
  text-align: center;
  margin: 1.5625rem 0;
}
.searchErrorWrapper__h06092 {
  padding-top: 3.75rem;
}
",
}
`;
